src/web/demo/src/main/java/org/geoserver/web/demo/CommonFormatLink.java

```diff
@@ -0,0 +1,40 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.web.demo;
+
+import org.apache.wicket.markup.html.link.ExternalLink;
+import org.geoserver.web.ToolLinkExternalInfo;
+
+public abstract class CommonFormatLink extends ToolLinkExternalInfo
+        implements Comparable<CommonFormatLink> {
+
+    private final String componentId = "theLink";
+    private int order = 1000;
+
+    public CommonFormatLink() {
+        super();
+    }
+
+    public abstract ExternalLink getFormatLink(
```

Comment:
Abstract methods also need documentation, plugin implementors need to implement them.

I'm worried about the "wmsVisible, wfsVisible" bits, what if we want to extend the preview page later, we'd have to break this method signature. Maybe best to expose the layer contained in PreviewLayer instead, and have the extension point compute the visibility on its own instead?  The MapPreviewPage.hasServiceSupport method could be moved into the WMS facade class for reuse in both places.

Reply:
I'll have a go at that

Reply:
What is the WMS facade class?

Reply:
I moved hasServiceSupport into PreviewLayer, if that is what you meant.  Seems to work ok, and those parameters are no longer required.  I'll push it and you can tell me what you think.